aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogTable.java

```diff
@@ -132,6 +134,28 @@ public void testCreateTableBadName() {
                 TableIdentifier.of(namespace, "table-1"), schema, partitionSpec));
   }
 
+  @Test
+  public void testCreateAndLoadTableWithoutWarehouseLocation() {
+    GlueCatalog glueCatalogWithoutWarehouse = new GlueCatalog();
+    glueCatalogWithoutWarehouse.initialize(
+        catalogName,
+        null,
+        new AwsProperties(),
+        glue,
+        LockManagers.defaultLockManager(),
+        new S3FileIO(clientFactory::s3),
+        ImmutableMap.of());
+    String namespace = createNamespace();
+    String tableName = getRandomName();
+    TableIdentifier identifier = TableIdentifier.of(namespace, tableName);
+    try {
+      glueCatalog.createTable(identifier, schema, partitionSpec, tableLocationProperties);
+      glueCatalog.loadTable(identifier);
+    } catch (RuntimeException e) {
+      fail("Create and load table without warehouse location should succeed");
```

Comment:
```suggestion
      throw new IOException("Create and load table without warehouse location should succeed", e);
```
We shouldn't hide the original exception. Without the exception message and the stacktrace, it's hard to find the root cause when an exception occur.

Reply:
yes, I was not able to find a `fail(...)` method that allows taking exception, thus this. Throwing IO exception seems too much, I changed the code to rethrow the runtime exception.